### PR TITLE
Add save button and spinner to locations editor

### PR DESCRIPTION
**Pull request by toBeOfUse**
This is, tragically, based on my earlier pull request, but that was semi-unavoidable as I had to have that shim to make Windows database logins work; I think this works as is, but it will, also tragically, require me to learn how to actually write tests for Vue components; luckily, I have good examples to model mine after in this codebase. Feel free to ignore this until the fate of the other PR is decided and I decide I've got tests in place here; just making this a PR so no one else takes this job.

### CODE REVIEW

**Comment by toBeOfUse** [comment]
image

File: package.json [hunk 1]
```diff
@@ -18,32 +18,35 @@
   },
   "dependencies": {
     "axios": "^0.19.0",
-    "bootstrap": "^4.2.1",
+    "bootstrap": "^4.4.1",
     "bootstrap-sass": "^3.4.0",
+    "eslint-plugin-prettier": "^3.1.2",
```
Comment (wlonk) [inline package.json]: This should be in `devDependencies` I think.

File: src/components/SaveButton.vue [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  },
+  props: {
+    allUploadsComplete: {
+      type: Boolean,
+      required: true,
+    },
+  },
+  computed: {
+    spinning() {
+      // making Absolutely Sure that both of these are accessed and thus registered as dependencies for the property, despite short-circuit evaluation
```
Comment (wlonk) [inline src/components/SaveButton.vue]: I prefer comments to be hard-wrapped to 80 characters, for reliable legibility.

File: src/components/SaveButton.vue [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+@keyframes spin {
+  0% {
+    -webkit-transform: rotate(0deg);
+    transform: rotate(0deg);
+  }
+  100% {
+    -webkit-transform: rotate(359deg);
+    transform: rotate(359deg);
+  }
+}
```
Comment (wlonk) [inline src/components/SaveButton.vue]: All this CSS work is delightful! I am very bad at CSS, thank you so much!

File: tests/unit/SaveButton.spec.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    expect(wrapper.vm.spinning).toBe(true);
+    wrapper.setProps({ allUploadsComplete: true });
+    expect(wrapper.vm.spinning).toBe(false);
+  });
+  test('Button is disabled when !allUploadsComplete', async () => {
+    wrapper.setProps({ allUploadsComplete: false });
+    const button = wrapper.find('button');
+    await wrapper.vm.$nextTick();
+    expect(button.element.disabled).toBe(true);
+  });
```
Comment (wlonk) [inline tests/unit/SaveButton.spec.js]: I'm OK with these tests, but as a general matter, I prefer more tests to tests that have a succession of state changes and expectations. It lets you more easily isolate test failures.

File: src/components/Identity.vue [hunk 5]
```diff
@@ -257,7 +257,10 @@ export default {
     update() {
       const url = window.Urls['api:identity-detail'](this.identity.id);
       const data = this.identity;
-      return this.$http.put(url, data);
+      const req = this.$http.put(url, data);
+      // emit the Promise created by the $http call so that the state of the upload can be tracked if so desired
+      this.$emit('upload-started', req);
```
Comment (wlonk) [inline src/components/Identity.vue]: An apparently small thing that I'm gonna be a bit of a hardass about: can you change all these cases of "upload" to "save"? "Upload" as a term implies file uploads to me very strongly, and as this is just POST/PUTting JSON to the server, it is _not that_. Without that name change, I will find this code hard to follow every time in the future that I revisit it.

File: src/components/IdentitiesForm.vue [hunk 6]
```diff
@@ -8,15 +8,21 @@
         :index="index"
         :disabled="identity.disabled"
         @destroy="destroy"
+        @upload-started="trackUpload"
```
Comment (wlonk) [inline src/components/IdentitiesForm.vue]: I love this use of a custom event.

File: src/components/Identity.vue (hunk 5 above)
Comment (toBeOfUse) [reply src/components/Identity.vue]: "Save" usually implies putting something directly on a local hard drive to me, and this is not that, but your point about file uploading is very fair. Kind of want to try out something silly like `transmission-started` then but I guess "save" should be fine

File: src/components/IdentitiesForm.vue (hunk 6 above)
Comment (toBeOfUse) [reply src/components/IdentitiesForm.vue]: thank you

File: src/components/Identity.vue (hunk 5 above)
Comment (wlonk) [reply src/components/Identity.vue]: "Transmission" is great! "Request" would work too?

I live in the web world enough that "save" usually means "to a remote" for me, so, "save" works for me, but I'm not _dedicated_ to it.

**Comment by toBeOfUse** [comment]
Feel free to make those changes; or, if you haven't happened to get around to them by this time tomorrow night or so I'm sure I'll go for it
